intermediate-js.md

```diff
@@ -3,37 +3,39 @@
 ### Without using Google answer the following:
 
 1. What is the difference between .map() and .filter()?
-
+Map and filter are both higher order functions.  The map function is used when you want to manipulate each item in an array while the filter function is used to extract specific data from an array.
 2. Why would you use object destructuring?
-
+Object destructuring is very handy because it allows you to essentially shorthand while writing new variables.
 3. What is the difference between var, let, and const?
-
+Var/let/const are all used to define variables.  "You should always use const when defining a variable, unless you have a good reason not to." -Matt.  Const will keep a variable the same, if you use let, you can redefine the variable again in the future.
 4. Why is testing important?
-
+Testing is important because you can make sure that all of your functions are working before you write a ton of code that won't work!
 5. What is a higher order function?
-
+A higher order function is one that either takes in another function or returns one (map,filter).
 6. What is the difference between a class and an object?
-
+Objects in JavaScript are essentially data storage while classes are declared in a constructor and called on in functions.
 7. What did you learn during the group project this week? Please include any additional feedback you may have.
-
+The group projects this week were awesome!  We worked really well together.  I learned a lot more about react, it was nice having two other partners.  It really paid off to have treasure hunt be essentially a dry run at tic tac toe.
 
 ### HTML/CSS Review questions: First, try to answer each question on your own then Google the answer to further your knowledge.
 
 1. How do you link a CSS file to your HTML page?
-
+It needs to be exported/imported, then individual elements can be called with href
 2. What is the difference between a div and a span?
-
+Span is usually used for inline styling while div is typically used for block.
 3. What is a CSS class? When should you use an id instead of a class?
-
+CSS classes are used to style an element.  I like to use an id if it's a more complicated style (using a button, some crazy animation, etc.)
 4. Name 4 semantic HTML tags.
-
+<h1><h2><h3><h4> ;) there's also <head><p><footer><header> and lots more!
```

Comment:
Nice start here! I'd like you to consider what the difference is between regular html and 'semantic' html